packages/unchained-client/src/evm/parser/types.ts

```diff
@@ -11,6 +11,7 @@ export enum TxParser {
 export interface ERC20TxMetadata extends BaseTxMetadata {
   parser: TxParser.ERC20
   assetId?: string
+  value?: string
```

Comment:
Coincidentally this metadata now resembles both yearn and erc20. I think we have a couple options:

- Make this a less strict `EvmTxMetadata` type that all evm parsers will use and we can then narrow by `isEvmParser` or something along those lines
- Only move `TxParser.Yearn` into here and figure out a reasonable name for the type to keep things more strict/accurate
- Update to have each parser define a `TxMetadata` type (either alias to standard or some specific fields), and union all parser metadatas together under the `evm.TxMetadata` umbrella

I feel like the 3rd option would be the most explicit and flexible pattern moving forward assuming all of the narrowing still works as intended, though there will be slight extra overhead keeping the union type up to date.

Reply:
I would be more inclined towards the first solution given the overhead of the last, but agree the third one is more accurate and should allow for better discrimination, will give it a try!

Reply:
Done in 598f7736693cc936f03265337657e98177f7fb0b - gave it a try while linked in web and `isTokenMetadata` still works and properly narrows TxMetadata type down with the new union :tada: